tests/st/utils/docker_host.py

```diff
@@ -250,7 +253,24 @@ def start_calico_node(self, options=""):
         args.append(options)
 
         cmd = ' '.join(args)
-        self.calicoctl(cmd)
+
+        if with_ipv4pool_cidr_env_var:
+            # Run the dryrun command, then modify and execute the command that
+            # that tells us.
+            output = self.calicoctl(cmd)
+            base_command = output.split('\n')[-4].rstrip()
```

Comment:
Agreed, I'll wait for #1511 

In case you're not already aware:
- I got this fragile approach from test_default_pools.py - so probably it would be good to update that also, once we have #1511 .
- It appeared to me during testing that there is quite a lot of difference between `calicoctl node run ...` and running the command that `calicoctl node run --dryrun ...` tells you to run.  At least there is a difference in the stdout:

calicoctl node run ... :

```
export ETCD_AUTHORITY=10.37.7.242:2379; export ETCD_SCHEME=http; export ETCD_CA_CERT_FILE=; export ETCD_CERT_FILE=; export ETCD_KEY_FILE=; /code/dist/calicoctl node run --node-image=calico/node --ip=10.37.7.242 --ip6=fe80::5b56:4bf5:d27c:4cd --no-default-ippools
  # Running command to load modules: modprobe -a xt_set ip6_tables
  # Enabling IPv4 forwarding
  # Enabling IPv6 forwarding
  # Increasing conntrack limit
  # Removing old calico-node container (if running).
  # Running the following command to start calico-node:
  # 
  # docker run --net=host --privileged --name=calico-node -d --restart=always -e CALICO_LIBNETWORK_IFPREFIX=cali -e IP=10.37.7.242 -e ETCD_AUTHORITY=10.37.7.242:2379 -e IP_AUTODETECTION_METHOD=first-found -e IP6_AUTODETECTION_METHOD=first-found -e NO_DEFAULT_POOLS=true -e CALICO_LIBNETWORK_LABEL_ENDPOINTS=false -e ETCD_ENDPOINTS= -e CALICO_NETWORKING_BACKEND=bird -e ETCD_SCHEME=http -e CALICO_LIBNETWORK_CREATE_PROFILES=true -e IP6=fe80::5b56:4bf5:d27c:4cd -e NODENAME=smaug -e CALICO_LIBNETWORK_ENABLED=true -v /var/log/calico:/var/log/calico -v /var/run/calico:/var/run/calico -v /lib/modules:/lib/modules -v /run/docker/plugins:/run/docker/plugins -v /var/run/docker.sock:/var/run/docker.sock calico/node
  # 
  # Image may take a short time to download if it is not available locally.
  # Container started, checking progress logs.
  # 
  # Skipping datastore connection test
  # Using IPv4 address from environment: IP=10.37.7.242
  # IPv4 address 10.37.7.242 discovered on interface eno1
  # Using IPv6 address from environment: IP6=fe80::5b56:4bf5:d27c:4cd
  # IPv6 address fe80::5b56:4bf5:d27c:4cd discovered on interface eno1
  # No AS number configured on node resource, using global value
  # Using node name: smaug
  # Starting libnetwork service
  # Calico node started successfully
```

calicoctl node run --dryrun ... and then executing that command:

```
export ETCD_AUTHORITY=10.37.7.242:2379; export ETCD_SCHEME=http; export ETCD_CA_CERT_FILE=; export ETCD_CERT_FILE=; export ETCD_KEY_FILE=; /code/dist/calicoctl node run --dryrun --node-image=calico/node
  # Use the following command to start the calico/node container:
  # 
  # docker run --net=host --privileged --name=calico-node -d --restart=always -e CALICO_LIBNETWORK_LABEL_ENDPOINTS=false -e ETCD_AUTHORITY=10.37.7.242:2379 -e ETCD_SCHEME=http -e NODENAME=smaug -e CALICO_NETWORKING_BACKEND=bird -e NO_DEFAULT_POOLS= -e CALICO_LIBNETWORK_ENABLED=true -e CALICO_LIBNETWORK_CREATE_PROFILES=true -e ETCD_ENDPOINTS= -e IP_AUTODETECTION_METHOD=first-found -e IP6_AUTODETECTION_METHOD=first-found -e CALICO_LIBNETWORK_IFPREFIX=cali -v /var/log/calico:/var/log/calico -v /var/run/calico:/var/run/calico -v /lib/modules:/lib/modules -v /run/docker/plugins:/run/docker/plugins -v /var/run/docker.sock:/var/run/docker.sock calico/node
  # 
  # If you are running calico/node in an init system, use the --init-system flag
  # to display the appropriate start and stop commands.
docker run --net=host --privileged --name=calico-node -d --restart=always -e CALICO_IPV4POOL_CIDR=10.0.0.0/27 -e CALICO_LIBNETWORK_LABEL_ENDPOINTS=false -e ETCD_AUTHORITY=10.37.7.242:2379 -e ETCD_SCHEME=http -e NODENAME=smaug -e CALICO_NETWORKING_BACKEND=bird -e NO_DEFAULT_POOLS= -e CALICO_LIBNETWORK_ENABLED=true -e CALICO_LIBNETWORK_CREATE_PROFILES=true -e ETCD_ENDPOINTS= -e IP_AUTODETECTION_METHOD=first-found -e IP6_AUTODETECTION_METHOD=first-found -e CALICO_LIBNETWORK_IFPREFIX=cali -v /var/log/calico:/var/log/calico -v /var/run/calico:/var/run/calico -v /lib/modules:/lib/modules -v /run/docker/plugins:/run/docker/plugins -v /var/run/docker.sock:/var/run/docker.sock calico/node
  # 53fdd27bf3ec9f68dc5ed3af8f905926378c88528bf5274cda52cbd9b871ca36
```

The difference in the stdout makes me wonder/worry if there is also a difference in what they _do_ - e.g. does the `--dryrun` version actually do the modprobe, enabling forwarding, etc.?